src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,201 @@
+import React, {Component} from 'react';
+import Hark from 'hark';
+import PropTypes from 'prop-types';
+import Graphicequaliser from './canvasequalizer/canvasequalizer';
+import PlayButton from './playbutton/playbutton';
+import Uploadbutton from './upload/uploadbutton';
+import Infoabouttrack from './upload/infoaboutfile/infoaboutuploadfile';
+import Streambutton from './streambutton/streambutton';
+import {connect} from 'react-redux';
+import {createAudioData, playPauseSoundFromFile, createStreamData, startMuteStreamAudio, mergeCanvasWidth} from '../../actions/audioactions';
+
+class Equalizer extends Component {
+  state = {
+    analyser: this.props.audioData.analyser,
```

Comment:
Are you need this in state or it could be just prop?
No need to create state with undefined values I can add them later

Reply:
yes need, there is a function with recursion and it need every cycle some data. Before i made big function for this, than class, now Lubomyr said to make local state. About keys with undefined it easier to me to see what state with what keys I have. I can delete them before prod.